contracts/facets/EmbalmerFacet.sol

```diff
@@ -121,4 +170,152 @@ contract EmbalmerFacet {
+    ///     contract and are ready to work.
+    ///   - Provide the arweave transaction id to be stored on chain.
+    ///   - Reward the archaeologist who uploaded to payload to arweave with the storage fee.
+    ///
+    /// @dev The archaeologistSignatures must be sent in the same order that the
+    /// archaeologists were sent to the initializeSarcophagus function,
+    /// otherwise the transaction will revert.
+    /// @param identifier the identifier of the sarcophagus
+    /// @param archaeologistSignatures the signatures of the archaeologists.
+    /// This is archaeologist.length - 1 since the arweave archaeologist will be providing their own signature.
+    /// @param arweaveArchaeologistSignature the signature of the archaeologist who uploaded to arweave
+    /// @param arweaveTxId the arweave transaction id
+    /// @param sarcoToken The erc20 sarcophagus token
+    /// @return The boolean true if the operation was successful
+    function finalizeSarcophagus(
+        bytes32 identifier,
+        LibTypes.SignatureWithAccount[] memory archaeologistSignatures,
+        LibTypes.Signature memory arweaveArchaeologistSignature,
+        string memory arweaveTxId,
+        IERC20 sarcoToken
+    ) external returns (bool) {
+        // Confirm that the sarcophagus exists
+        if (
+            s.sarcophaguses[identifier].state !=
+            LibTypes.SarcophagusState.Exists
+        ) {
+            revert LibErrors.SarcophagusDoesNotExist(identifier);
+        }
+
+        // Confirm that the embalmer is making this transaction
+        if (s.sarcophaguses[identifier].embalmer != msg.sender) {
+            revert LibErrors.SenderNotEmbalmer(
+                msg.sender,
+                s.sarcophaguses[identifier].embalmer
+            );
+        }
+
+        // Confirm that the sarcophagus is not already finalized by checking if
+        // the arweaveTxId is empty
+        if (bytes(s.sarcophaguses[identifier].arweaveTxId).length > 0) {
+            revert LibErrors.SarcophagusAlreadyFinalized(identifier);
+        }
+
+        // Confirm that the provided arweave transaction id is not empty
+        if (bytes(arweaveTxId).length == 0) {
+            revert LibErrors.ArweaveTxIdEmpty();
+        }
+
+        // Confirm that the correct number of archaeologist signatures was sent
+        // This will be archaeologist.length - 1 since the arweave archaeoligist
+        // will be providing their own signature.
+        if (
+            archaeologistSignatures.length !=
+            s.sarcophaguses[identifier].archaeologists.length - 1
+        ) {
+            revert LibErrors.IncorrectNumberOfArchaeologistSignatures(
+                archaeologistSignatures.length
+            );
+        }
+
+        // Iterate over each regular archaeologist signature. This will not
+        // include the arweave archaeologist.
+        for (uint256 i = 0; i < archaeologistSignatures.length; i++) {
```

Comment:
The only extra validation to consider would be the uniqueness of the signatures. Could theoretically pass in the same signature 4 times here, and this wouldn't revert.

Reply:
Yet another astute observation. I'll have to think about the best way to do this.